File: C++/Jump-Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxlen = 0, cur = 0;
        while (cur < nums.size() && cur <= maxlen) {
        	maxlen = max(maxlen, cur + nums[cur]);
        	if (maxlen >= nums.size() - 1) return true;
        	cur ++;
        }
        return false;
    }
};

File: Python/Jump-Game.py
#!/usr/bin/env python


def canJump(self, A):
    lenth = len(A) - 1
    if lenth <= 0:
        return True
    maxlenth = A[0]
    for i in range(lenth+1):
        if i <= maxlenth:
            maxlenth = max(maxlenth, A[i] + i)
        else:
            return False
        if maxlenth >= lenth:
            return True